presubmit-tests.sh

```diff
@@ -138,42 +138,18 @@ function default_build_test_runner() {
   [[ -z "${go_pkg_dirs}" ]] && return ${failed}
   # Ensure all the code builds
   subheader "Checking that go code builds"
-  local report
-  report="$(mktemp)"
-  local errors_go1=""
-  local errors_go2=""
-  if ! capture_output "${report}" go build -v ./... ; then
-    failed=1
-    # Consider an error message everything that's not a package name.
-    errors_go1="$(grep -v '^\(github\.com\|knative\.dev\)/' "${report}" | sort | uniq)"
-  fi
   # Get all build tags in go code (ignore /vendor, /hack and /third_party)
   local tags
-  tags="$(grep -r '// +build' . \
-    | grep -v '^./vendor/' | grep -v '^./hack/' | grep -v '^./third_party' \
-    | cut -f3 -d' ' | tr ',' '\n' | sort | uniq | tr '\n' ' ')"
-  local tagged_pkgs
-  tagged_pkgs="$(grep -r '// +build' . \
-    | grep -v '^./vendor/' | grep -v '^./hack/' | grep -v '^./third_party' \
-    | grep ":// +build " | cut -f1 -d: | xargs dirname \
-    | sort | uniq | tr '\n' ' ')"
-  for pkg in ${tagged_pkgs}; do
-    # `go test -c` lets us compile the tests but do not run them.
-    if ! capture_output "${report}" go test -c -tags="${tags}" "${pkg}" ; then
-      failed=1
-      # Consider an error message everything that's not a successful test result.
-      errors_go2+="$(grep -v '^\(ok\|\?\)\s\+\(github\.com\|knative\.dev\)/' "${report}")"
-    fi
-    # Remove unused generated binary, if any.
-    rm -f e2e.test
-  done
+  tags="$(find . \
+    -path './vendor' -prune -o -path './hack' -prune -o -path './third_party' -prune \
+    -o -type f -name '*.go' -exec grep '// +build' {} + \
```

Comment:
we probably want to include go1.17 build tag format as well

Reply:
https://go.googlesource.com/proposal/+/master/design/draft-gobuild.md

Reply:
It is not required at this point. But, it will be in the future.

In fact, I'm writing a Go tool to simplify this, and also be included in knobots:

```
tags="$(go run knative.dev/test-infra/tools/go-ls-tags@latest \
  --exclude vendor --exclude third_party --exclude hack)"
```

But, I think that can be delivered with separate PRs

Reply:
Here's a draft PR of that tool: https://github.com/knative/test-infra/pull/2995